src/views.js

```diff
@@ -438,13 +438,23 @@ const AttachmentChatboxMessageView = new Lang.Class({
         let thumbnailFactory = Thumbnailer.forSize(GnomeDesktop.DesktopThumbnailSize.LARGE);
         let preview = getPreviewForFile(this.state.path, thumbnailFactory);
         if (preview.thumbnail) {
+            this.attachment_contents.orientation = Gtk.Orientation.VERTICAL;
+
             // Where we are using a thumbnail, use GtkCSSProivder to
             // set the background image
+            let naturalWidth = this.get_preferred_size()[1].width;
+            let minWidth = preview.thumbnail.width < naturalWidth ? naturalWidth : preview.thumbnail.width;
```

Comment:
OK... in that case I think you can still remove the comparison to the natural width requisition, and just use the preview thumbnail dimensions to compute the aspect ratio.